crates/sncast/src/starknet_commands/deploy.rs

```diff
@@ -92,35 +92,45 @@ pub async fn deploy(
         l1_data_gas_price,
     } = fee_settings.expect("Failed to convert to fee settings");
 
-    let execution = match l1_gas {
-        None => execution,
-        Some(l1_gas) => execution.l1_gas(l1_gas),
-    };
-    let execution = match l1_gas_price {
-        None => execution,
-        Some(l1_gas_price) => execution.l1_gas_price(l1_gas_price),
-    };
-    let execution = match l2_gas {
-        None => execution,
-        Some(l2_gas) => execution.l2_gas(l2_gas),
-    };
-    let execution = match l2_gas_price {
-        None => execution,
-        Some(l2_gas_price) => execution.l2_gas_price(l2_gas_price),
-    };
-    let execution = match l1_data_gas {
-        None => execution,
-        Some(l1_data_gas) => execution.l1_data_gas(l1_data_gas),
-    };
-    let execution = match l1_data_gas_price {
-        None => execution,
-        Some(l1_data_gas_price) => execution.l1_data_gas_price(l1_data_gas_price),
-    };
-    let execution = match nonce {
-        None => execution,
-        Some(nonce) => execution.nonce(nonce),
-    };
-    let result = execution.send().await;
+    let deployment = apply_optional_fields!(
+        deployment,
+        l1_gas => DeploymentV3::l1_gas,
+        l1_gas_price => DeploymentV3::l1_gas_price,
+        l2_gas => DeploymentV3::l2_gas,
+        l2_gas_price => DeploymentV3::l2_gas_price,
+        l1_data_gas => DeploymentV3::l1_data_gas,
+        l1_data_gas_price => DeploymentV3::l1_data_gas_price,
+        nonce => DeploymentV3::nonce
```

Comment:
Assuming that the commented out code was just condensed to that?

Reply:
Yup, left the commented out one by accident.